[*] Add `CONTRIBUTING.md` file (#198)
MaximilianAnzinger/pgdp2223-tests: CONTRIBUTING.md

# Contributing

You want to contribute and add tests to this repository? Great! We are happy to
accept your contributions. Please read the following guidelines before you
start.

## Guidelines for contributing

**TL;DR**

* PR title and description should use a prefix like `[W09H03] Add behavioral tests`
* Separate your tests logically into small files
* Don't add tests that are not needed for the exercise
* Write deterministic tests
* Use JUnit 5.9.1
* Format your code

> * Don't depend on changing the code the implementation

Reviewer: ```suggestion
* Don't depend on changing the code of the implementation
```

* Every test file should end with `Test.java`
* Follow the package structure of the implementation
* Only one test case per test method

> * Use descriptive names for test methods.

Reviewer 2: ```suggestion
* Use descriptive names for test methods.
* As much as possible, follow the [Java Style Guide](https://google.github.io/styleguide/javaguide.html) for variable and class notation.
```

Reviewer 3: "Follow the Java Style Guide for variable and class notation as best as possible" sounds better imo.

Author: I have replaced "Format your code" with "Follow the Java Style Guide"

### PR Title and Description

#### Title

Please as prefix for your PR title the name of the exercise you are working on.

```
[W09H03] Add behavioral tests
```

If you don't changed the tests for an exercise, you can use `[*]` as prefix:

```
[*] ...
```

#### Description

> In the description of your PR, please add a description that helps us to understand what you are doing. For example, when you fix a bug, please add a description of the bug and how you fixed it.

Reviewer 2: ```suggestion
The description of your PR should help us understand what you are doing: For example, when fixing a bug, provide information on the bug and a brief sentence on how you fixed it.
```
not good to repeat long words in the same sentence imo

Author: Updated ✅

### Don't add tests that are not needed for the exercise

Please don't add tests that are not needed for the exercise. For example, if the
exercise only requires you to implement a method `foo(int a)` and the exercise
says that `a` is never negative, you don't need to test if `a` is negative.

### Write deterministic tests

Please make sure that your tests are deterministic. For example, if you test a
method that returns a random number, you should use a fixed seed for the
random number generator.

**BAD**:
```java
Random random = new Random();
```

**GOOD**:
```java
// Use a fixed seed for the random number generator
Random random = new Random(42);
```

### Use JUnit 5.9.1

We use JUnit 5.9.1 for testing. You can find the documentation
[here](https://junit.org/junit5/docs/current/user-guide/).

To use JUnit 5.9.1 in your project, you need to add the following dependency to your `build.gradle`.

```gradle
dependencies {
    testImplementation 'org.junit.jupiter:junit-jupiter:5.9.1'
}
```

### Format your code

Make sure that your code is formatted according to the [Google Java Style
Guide](https://google.github.io/styleguide/javaguide.html).

> ### Don't depend on changing the code the implementation

Author: ```suggestion
### Don't depend on changing the code of the implementation
```
